packages/react/src/components/Comments/Comment.tsx

```diff
@@ -0,0 +1,341 @@
+"use client";
+
+import { CommentData, ThreadData, mergeCSSClasses } from "@blocknote/core";
+import {
+  ComponentPropsWithoutRef,
+  MouseEvent,
+  ReactNode,
+  useCallback,
+  useState,
+} from "react";
+import {
+  RiArrowGoBackFill,
+  RiCheckFill,
+  RiDeleteBinFill,
+  RiEditFill,
+  RiEmotionLine,
+  RiMoreFill,
+} from "react-icons/ri";
+
+import { useComponentsContext } from "../../editor/ComponentsContext.js";
+import { useBlockNoteEditor } from "../../hooks/useBlockNoteEditor.js";
+import { useCreateBlockNote } from "../../hooks/useCreateBlockNote.js";
+import { useDictionary } from "../../i18n/dictionary.js";
+import { CommentEditor } from "./CommentEditor.js";
+import { EmojiPicker } from "./EmojiPicker.js";
+import { ReactionBadge } from "./ReactionBadge.js";
+import { schema } from "./schema.js";
+import { useUser } from "./useUsers.js";
+
+export interface CommentProps extends ComponentPropsWithoutRef<"div"> {
+  /**
+   * The comment to display.
+   */
+  comment: CommentData;
+
+  /**
+   * The thread the comment belongs to.
+   */
+  thread: ThreadData;
+
+  /**
+   * How to show or hide the actions.
+   */
+  showActions?: boolean | "hover";
+
+  /**
+   * Whether to show the resolve action.
+   */
+  showResolveAction?: boolean;
+
+  /**
+   * Whether to show the comment if it was deleted. If set to `false`, it will render deleted comments as `null`.
+   */
+  showDeleted?: boolean;
+
+  /**
+   * Whether to show reactions.
+   */
+  showReactions?: boolean;
+}
+
+/**
+ * The Comment component displays a single comment with actions,
+ * a reaction list and an editor when editing.
+ *
+ * It's generally used in the `Thread` component for comments that have already been created.
+ *
+ */
+export const Comment = ({
```

Comment:
This component is doing an awful lot. Consider breaking this up into either multiple components, or extract a hook to keep the business logic separate.